### PR TITLE
Remove Unused Types

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change ###

Remove NavigationMenu & Renderers
 - Never completed and there is the new CollectionView
 - Only (partially) implemented on iOS/Android
 - Internal
 
Removing ToolbarButton and related types
 - `ToolbarButton`, `ToolbarImageButton`, `IToolbarButton`
 - this appears not to be used at all
 - was part of `ToolbarRenderer` in the old Duplo repo, which has since been removed

### Issues Resolved ### 
 - housekeeping

### API Changes ###

 Removed:
 - Core
    - `NavigationMenu`
 - Android
    - `NavigationMenuRenderers`
    - `ToolbarButton`
    - `ToolbarImageButton`
    - `IToolbarButton`
- iOS
    - `NavigationMenuRenderers`
 

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- Android

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

None

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by samhouts** [comment]
build --uitests

File: Stubs/Xamarin.Forms.Platform.cs [hunk 1]
```diff
@@ -105,15 +105,6 @@ internal class _ActivityIndicatorRenderer { }
...
 	internal class _FrameRenderer { }
 
-#if !WINDOWS_PHONE && !WINDOWS_PHONE_APP && !TIZEN4_0
-	[RenderWith (typeof (NavigationMenuRenderer))]
-	internal class _NavigationMenuRenderer { }
-
-	[RenderWith (typeof (OpenGLViewRenderer))]
-#else
-	[RenderWith (null)]
-#endif
```
Comment (StephaneDelcroix) [inline Stubs/Xamarin.Forms.Platform.cs]: you're dropping too much here.

Comment (mattleibow) [reply Stubs/Xamarin.Forms.Platform.cs]: Whoops! Fixed.

**Comment by samhouts** [comment]
build --uitests

**Pull request merged**